src/Application/Migrations/15_function_ticket_state.sql

```diff
@@ -19,7 +19,16 @@ BEGIN
 	WHERE
 		pts.project_id = param_project_id AND
 		ts2.ticket_type = param_ticket_type AND
-		ts2.ticket_state = param_ticket_state
+		ts2.ticket_state = param_ticket_state AND
+		(pts.skip_on_dependent = false OR
+			( /* is master encoding ticket */
+				SELECT ep.depends_on
```

Comment:
Like with other properties before, for performance improvements a column "is_master" in tbl_ticket filled on INSERT e.g. using a trigger might be useful.